src/test/java/com/michelin/ns4kafka/controller/SchemaControllerTest.java

```diff
@@ -474,23 +499,126 @@ void shouldDeleteSchema() {
         doNothing().when(applicationEventPublisher).publishEvent(any());
 
-        StepVerifier.create(schemaController.delete("myNamespace", "prefix.subject-value", false))
+        StepVerifier.create(schemaController.delete("myNamespace", "prefix.subject-value", Optional.empty(), false))
+            .consumeNextWith(response -> assertEquals(HttpStatus.NO_CONTENT, response.getStatus()))
+            .verifyComplete();
+
+        verify(schemaService).deleteAllVersions(namespace, "prefix.subject-value");
+    }
+
+    @Test
+    void shouldDeleteOldSchemaVersionAndNotSendEventLog() {
+        Namespace namespace = buildNamespace();
+        Schema schema1 = buildSchema();
+        Schema schema2 = buildSchemaV2();
+
+        when(namespaceService.findByName("myNamespace"))
+            .thenReturn(Optional.of(namespace));
+        when(schemaService.isNamespaceOwnerOfSubject(namespace, "prefix.subject-value"))
+            .thenReturn(true);
+        when(schemaService.getSubject(namespace, "prefix.subject-value", "1"))
+            .thenReturn(Mono.just(schema1));
+        when(schemaService.deleteVersion(namespace, "prefix.subject-value", "1"))
+            .thenReturn(Mono.just(Optional.of(schema2)));
+
+        StepVerifier.create(schemaController.delete("myNamespace", "prefix.subject-value", Optional.of("1"), false))
+            .consumeNextWith(response -> assertEquals(HttpStatus.NO_CONTENT, response.getStatus()))
+            .verifyComplete();
+
+        verify(applicationEventPublisher, never()).publishEvent(any());
+    }
+
+    @Test
+    void shouldDeleteLatestSchemaVersionBySpecifyingLatest() {
+        Namespace namespace = buildNamespace();
+        Schema schema1 = buildSchema();
+        Schema schema2 = buildSchemaV2();
+
+        when(namespaceService.findByName("myNamespace"))
+            .thenReturn(Optional.of(namespace));
+        when(schemaService.isNamespaceOwnerOfSubject(namespace, "prefix.subject-value"))
+            .thenReturn(true);
+        when(schemaService.getSubject(namespace, "prefix.subject-value", "latest"))
+            .thenReturn(Mono.just(schema2));
+        when(schemaService.deleteVersion(namespace, "prefix.subject-value", "latest"))
+            .thenReturn(Mono.just(Optional.of(schema1)));
+        when(securityService.username())
+            .thenReturn(Optional.of("test-user"));
+        when(securityService.hasRole(ResourceBasedSecurityRule.IS_ADMIN))
+            .thenReturn(false);
+        doNothing().when(applicationEventPublisher).publishEvent(any());
+
+        StepVerifier.create(schemaController.delete("myNamespace", "prefix.subject-value",
+            Optional.of("latest"), false))
             .consumeNextWith(response -> assertEquals(HttpStatus.NO_CONTENT, response.getStatus()))
             .verifyComplete();
 
-        verify(schemaService).delete(namespace, "prefix.subject-value");
+        verify(applicationEventPublisher, times(1)).publishEvent(any());
```

Comment:
You can remove `times(1)` and the associated import, this is the default value used by `verify`

Reply:
OK